### DISCUSSION TITLE
Streaming Metrics

### DISCUSSION THREAD

**Discussion by pndhd1** [pedroSG94/RootEncoder#1572] [Q&A] [answered]
Hi Pedro. Which metrics can I obtain from the library? For example, are packet loss, bandwidth, RTT, frame rate available for monitoring streaming performance?

**Comment by pedroSG94** [comment, marked as answer]
Hello,

You can obtain all that info mentioned about stream except RTT.
From StreamClient (getStreamClient method) class you can obtain this info:
```kotlin
  abstract fun hasCongestion(): Boolean
  abstract fun getCacheSize(): Int
  abstract fun getItemsInCache(): Int
  abstract fun getSentAudioFrames(): Long
  abstract fun getSentVideoFrames(): Long
  abstract fun getDroppedAudioFrames(): Long
  abstract fun getDroppedVideoFrames(): Long
```
Using callbacks of ConnectChecker you can get the bandwidth from:
```kotlin
fun onNewBitrate(bitrate: Long)
```
About fps, you can get fps using setFpsListener method and listening the callback provided (this fps include dropped frames).

I'm not sure if calculate RTT in java sockets is possible (normally do a ping could work but the server can disable a ping request). If you know a way I can implement it.

**Reply by pndhd1** [reply]
Thank you for the detailed information! For now, we are conducting an analysis. If we need RTT, I will create a separate feature request.
